### DISCUSSION TITLE
Custom DNS with client groups

### DISCUSSION THREAD

**Discussion by actioninja** [0xERR0R/blocky#1466] [Q&A]
Upfront info to avoid an XY Problem:
I have an internally hosted server that I would like accessible from the greater internet, as well as from a VPN, and from internal traffic. I currently use Cloudflare's proxied DNS provider for greater internet access. This works for most usecases, but causes problems on occasion, as well as makes configuration of IP based whitelists clunkier and less intuitive. An easy way of avoiding the cloudflare proxy for VPN and internal traffic is a local DNS provider that resolves the domains in question before they go upstream to the correct internal/vpn IP. However, the VPN ips and local network IPs are different.

I would like to be able to use a single instance of Blocky to be able to handle VPN traffic as well as internal network traffic, however the only way to do this seems to be to run two instances, one for the VPN and one for the internal network. Is there a way to do this or am I going to be stuck using two servers?

**Comment by kwitsch** [comment]
Not that I'm aware of.

But what's the problem running two instances of blocky on one server? 
You could assign multiple virtual IP addresses to your network card and make each instance listen to one. 
For less redundant configs one instance could use the other as upstream and forward the client IP through ECS.
